app/main.py

```diff
@@ -183,7 +185,7 @@ def load_mock_data() -> tuple[list[DatasetMetadata], dict[UUID, UnitData]]:
             for path in survey_mock_data_path.iterdir():
                 dataset_id = generate_dataset_id(
                     survey_id=survey_id,
-                    schema_version=SCHEMA_VERSION,
+                    schema_version=get_schema_version(path.parent.name, path.name),
```

Comment:
Could use the file stem now instead, as the example files are named corresponding to their schema version?

Reply:
Would that not have an impact with the test schema folders with v3.json?

Reply:
Ah, I see what you mean about the test `v3.json`. To get this working for now, we could update `v3.json` to have `"schema_version": "v3"`, because that's the pattern we're using in the business example files. Although, we don't have a schema `v3` yet, so it may be a bit confusing but it's a mock service and might not matter as much.

The original example test files represented different **dataset versions**, rather than different **schema versions** like they are now. For example, [this was how the mock dataset_metadata response used to look](https://github.com/ONSdigital/eq-runner-mock-sds/blob/ef0bde207a664e5059dae5e19081c50b11ed9ed9/mock_data/supplementary_dataset_metadata_response.json), where each file corresponded to different datasets but for **one schema version**

Unrelated to your ticket, but we should revisit & rejig these example test files for consistency with the example data in the SDS schema definitions repo. We can bolt on the unit data separately then, e.g. v1.json could have 3 dataset examples, v2.json could have 5 dataset examples etc. I'll pop a ticket up for this.

Hope I've explained this okay, it's probably easier to explain over a call. And sorry for the wall of text 😆

Reply:
As discussed in the tech session we'd keep the code as it is and create a "clean up" ticket to address the test data.